extensions/widgets/header/header.lcb

```diff
@@ -817,28 +960,217 @@ end handler
+      put fetchRect("searchIcon") into tSearchRect
    else
       put rectangle [0,0,0,0] into tSearchRect
    end if
-
    return tSearchRect
 end handler
 
 private handler locToAction(in pLoc as Point) returns Integer
    variable tX as Integer
-
    repeat with tX from 1 up to the number of elements in mActionRects
       if pLoc is within (element tX of mActionRects) then
          return (tX + 2)
       end if
    end repeat
-
    return 0
 end handler
 
+-- this handler returns an array of the default action
+private handler defaultActionsArray() returns Array
+   variable tArray as Array
+   put the empty array into tArray
+   put "action" into tArray["name"]
+   put "Action" into tArray["label"]
+   put "circle"into tArray["icon_name"]
+   return tArray
+end handler
+--
+
+-- this handler checks whether the android theme is dark or light based on the color scheme
+private handler checkIfThemeIsDark() returns nothing
+   variable tDarkColorSchemes as List
+   put [kDarkColor,kRedColor,kPinkColor,kPurpleColor,kDeepPurpleColor,kIndigoColor,kBlueColor,kTealColor,kDeepOrangeColor,kBrownColor,kBlueGreyColor] into tDarkColorSchemes
+
+   if mIsOpaque then
+      if mColorScheme is in tDarkColorSchemes then
+         put color [1,1,1] into mAndroidTitleColor
+         if mTitleIsDistinct then
+            put color [0,0,0,0.3] into mAndroidIconColor
+         else
+            put color [1,1,1] into mAndroidIconColor
+         end if
+      else
+         put color [0,0,0] into mAndroidTitleColor
+         if mTitleIsDistinct then
+            put color [1,1,1,0.38] into mAndroidIconColor
+         else
+            put color [0,0,0,0.87] into mAndroidIconColor
+         end if
+      end if
+   else
+      put color [0,0,0] into mAndroidTitleColor
+      put color [0,0,0,0.87] into mAndroidIconColor
+   end if
+
+end handler
+--
+
+private handler fetchAndroidColors() returns nothing
+	put the empty array into mAndroidColors
+   put "250,250,250" into mAndroidColors["Light"]
+   put "39,39,39" into mAndroidColors["Dark"]
+	put "238,43,41" into mAndroidColors["Red"]
+	put "224,0,81" into mAndroidColors["Pink"]
+	put "137,0,161" into mAndroidColors["Purple"]
+	put "83,33,168" into mAndroidColors["Deep Purple"]
+	put "48,58,165" into mAndroidColors["Indigo"]
+	put "30,128,240" into mAndroidColors["Blue"]
+	put "20,150,241" into mAndroidColors["Light Blue"]
+	put "22,175,202" into mAndroidColors["Cyan"]
+	put "17,133,117" into mAndroidColors["Teal"]
+	put "63,164,63" into mAndroidColors["Green"]
+	put "122,186,58" into mAndroidColors["Light Green"]
+	put "194,215,45" into mAndroidColors["Lime"]
+	put "254,233,46" into mAndroidColors["Yellow"]
+	put "253,182,13" into mAndroidColors["Amber"]
+	put "253,133,9" into mAndroidColors["Orange"]
+	put "251,63,28" into mAndroidColors["Deep Orange"]
+	put "101,67,56" into mAndroidColors["Brown"]
+	put "140,140,140" into mAndroidColors["Grey"]
+	put "78,106,120" into mAndroidColors["Blue Grey"]
+end handler
```

Comment:
You should probably use the named color constants for the keys here just to make sure everything is being checked correctly